server/routes/visitors.ts

```diff
@@ -123,5 +137,79 @@ export default function routes(
     }
   )
 
+  router.get(
+    '/select-main-contact/:offenderNo',
+    param('offenderNo').custom((value: string) => {
+      if (!isValidPrisonerNumber(value)) {
+        throw new Error('Invalid prisoner number supplied')
+      }
+
+      return true
+    }),
+    async (req, res) => {
+      const { offenderNo } = req.params
+      const errors = validationResult(req)
+
+      res.render('pages/mainContact', {
+        errors: !errors.isEmpty() ? errors.array() : [],
+        offenderNo,
+        adultVisitors: req.session.adultVisitors,
+      })
+    }
+  )
+
+  router.post(
+    '/select-main-contact/:offenderNo',
+    body('contact').custom((value: string) => {
+      if (!value) {
+        throw new Error('No main contact selected')
+      }
+
+      return true
+    }),
+    body('someoneElseName').custom((value: string, { req }) => {
+      if (value === '' && req.body.contact === 'someoneElse') {
+        throw new Error('Enter the name of the main contact')
+      }
+
+      return true
+    }),
+    body('phoneNumber').custom((value: string) => {
+      if (!value) {
+        throw new Error('Enter a phone number')
+      }
+
+      if (!/^(?:0|\+?44)(?:\d\s?){9,10}$/.test(value)) {
+        throw new Error('Enter a valid UK phone number, like 01632 960 001, 07700 900 982 or +44 808 157 0192')
+      }
+
+      return true
+    }),
+    param('offenderNo').custom((value: string) => {
+      if (!isValidPrisonerNumber(value)) {
+        throw new Error('Invalid prisoner number supplied')
+      }
+
+      return true
+    }),
+    async (req, res) => {
+      const { offenderNo } = req.params
+      const errors = validationResult(req)
+
+      if (!errors.isEmpty()) {
+        return res.render('pages/mainContact', {
+          errors: !errors.isEmpty() ? errors.array() : [],
+          offenderNo,
+          adultVisitors: req.session.adultVisitors,
+          phoneNumber: req.body.phoneNumber,
```

Comment:
We'll need this new data in the session too? Though could be done as part of the next page....

Reply:
I'll have a nosey through this tomorrow 🙂